aeron-system-tests/src/test/java/io/aeron/cluster/ClusterTest.java

```diff
@@ -1632,6 +1641,384 @@ public void shouldRecoverWhenFollowersIsMultipleTermsBehindFromEmptyLogAndPartia
         cluster.assertRecordingLogsEqual();
     }
 
+    @Test
+    @InterruptAfter(10)
+    void shouldRejectTakeSnapshotRequestWithAnAuthorisationError()
+    {
+        cluster = aCluster().withStaticNodes(3).start();
+        systemTestWatcher.cluster(cluster);
+
+        final TestNode leader = cluster.awaitLeader();
+        final List<TestNode> followers = cluster.followers();
+
+        final long requestCorrelationId = System.nanoTime();
+        final MutableBoolean responseReceived = new MutableBoolean();
+        cluster.egressListener(new EgressListener()
```

Comment:
Could the EgressListener logic be moved over to the TestCluster or extracted elsewhere to avoid inlining the definition in the tests?

Reply:
Done.